pennylane/templates/subroutines/flip_sign.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the FlipSign template.
+"""
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class FlipSign(Operation):
+    r"""FlipSign operator flips the sign for a given state.
+
+    In a nutshell, this class perform the following operation:
+
+    FlipSign(n):math:`|m\rangle = -|m\rangle` if len(m) = len(n)
+    FlipSign(n):math:`|m\rangle = |m\rangle` if len(m) != len(n)
+
+    Where m is the state to flip and n is the array of qubits (wires).
+    It flips the sign of the state.
+
+    Args:
+        wires (array[int]): wires that the operator acts on
+        bin_arr (array[int]): binary array vector representing the state to flip the sign
+
+    Raises:
+        ValueError: "expected at integer binary array "
+        ValueError: "expected at integer binary array for wires "
+        ValueError: "expected at integer binary array not empty "
+        ValueError: "expected at least one wire representing the qubit "
+
+    .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+    .. details::
+
+        :title: Usage Details
+
+        The template is used inside a qnode.
+        The number of shots has to be explicitly set on the device when using sample-based measurements:
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=5, shots = 1000)
+
+            @qml.qnode(dev)
```

Comment:
the example is fine but the functionality is not visible 😄 Leave this comment for the moment, we will see it later but I am writing it so that we do not forget it

Reply:
OK, I leave this open until we find a better enhancement.